### ISSUE TITLE
clientssl annotation and cert/key loggin issue CIS 1.14

### ISSUE THREAD

**Issue by hunterx19** [F5Networks/k8s-bigip-ctlr#1233]
I have an issue after upgrading the controller to version 1.14.
Bunch of these Warning messages are logged every 30 seconds

```shell-session
2020/04/08 07:36:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:45 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:36:53 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:15 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
2020/04/08 07:37:23 [WARNING] Both clientssl annotation and cert/key provided for Route: jenkins-f5-edge, using annotation.
```

I am using these versions:
AS3 3.17.1
BIG-IP 14.1.2.2 build 0.0.4
oc v3.11.0+62803d0-1
kubernetes v1.11.0+d4cacc0
Problem is showing over 3 different OpenShift clusters ( two okd and one dose )

Controller config
```yaml 
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: '23'
  creationTimestamp: '2019-12-06T16:23:52Z'
  generation: 53
  labels:
    app: k8s-bigip-ctlr
  name: f5sidc2-test-ctlr
  namespace: kube-system
  resourceVersion: '74924892'
  selfLink: /apis/apps/v1/namespaces/kube-system/deployments/f5sidc2-test-ctlr
  uid: ca7c1ea5-1844-11ea-be31-005056aa2127
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: k8s-bigip-ctlr
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: k8s-bigip-ctlr
      name: k8s-bigip-ctlr
    spec:
      containers:
        - args:
            - '--bigip-username=$(BIGIP_USERNAME)'
            - '--bigip-password=$(BIGIP_PASSWORD)'
            - '--bigip-url=10.203.99.166'
            - '--bigip-partition=ostest'
            - '--pool-member-type=cluster'
            - '--openshift-sdn-name=/Common/octest-tunnel'
            - '--manage-routes=true'
            - '--trusted-certs-cfgmap=kube-system/f5sidc2-testcert'
            - '--log-as3-response=true'
            - '--route-http-vserver=http_octest'
            - '--route-https-vserver=https_octest'
            - '--route-vserver-addr=10.70.54.100'
            - '--route-label=f5route'
            - '--agent=as3'
          command:
            - /app/bin/k8s-bigip-ctlr
          env:
            - name: BIGIP_USERNAME
              valueFrom:
                secretKeyRef:
                  key: username
                  name: bigip-login
            - name: BIGIP_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: password
                  name: bigip-login
          image: 'f5networks/k8s-bigip-ctlr:1.14.0'
          imagePullPolicy: Always
          name: k8s-bigip-ctlr
          resources: {}
          terminationMessagePath: /dev/termination-log
          terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      imagePullSecrets:
        - name: bigip-login
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: bigip-ctlr
      serviceAccountName: bigip-ctlr
      terminationGracePeriodSeconds: 30
status:
  availableReplicas: 1
  conditions:
    - lastTransitionTime: '2019-12-06T16:23:52Z'
      lastUpdateTime: '2019-12-06T16:23:52Z'
      message: Deployment has minimum availability.
      reason: MinimumReplicasAvailable
      status: 'True'
      type: Available
    - lastTransitionTime: '2019-12-06T16:23:52Z'
      lastUpdateTime: '2020-04-07T19:55:52Z'
      message: ReplicaSet "f5sidc2-test-ctlr-76476bdd76" has successfully progressed.
      reason: NewReplicaSetAvailable
      status: 'True'
      type: Progressing
  observedGeneration: 53
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
```

Route

```yaml 
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    virtual-server.f5.com/balance: least-connections-member
    virtual-server.f5.com/clientssl: /Common/oc-test_default_route_clientssl
    virtual-server.f5.com/health: |
      [
        {
          "path": "jenkins-edge.f5.test-mars.card.pri/",
          "send": "HTTP GET /",
          "interval": 5,
          "timeout": 10
        }
      ]
  creationTimestamp: '2020-03-11T11:14:22Z'
  labels:
    app: jenkins-ephemeral
    f5type: f5route
  name: jenkins-f5-edge
  namespace: test-gorans
  resourceVersion: '74921124'
  selfLink: /apis/route.openshift.io/v1/namespaces/test-gorans/routes/jenkins-f5-edge
  uid: 75a440e2-6389-11ea-b119-005056aa2127
spec:
  host: jenkins-edge.f5.test-mars.card.pri
  port:
    targetPort: web
  tls:
    termination: edge
  to:
    kind: Service
    name: jenkins
    weight: 100
  wildcardPolicy: None
status:
  ingress:
    - conditions:
        - lastTransitionTime: '2020-03-11T11:14:22Z'
          status: 'True'
          type: Admitted
      host: jenkins-edge.f5.test-mars.card.pri
      routerName: router
      wildcardPolicy: None
    - conditions:
        - lastTransitionTime: '2020-04-05T12:46:42Z'
          status: 'True'
          type: Admitted
      host: jenkins-edge.f5.test-mars.card.pri
      routerName: F5 BIG-IP
      wildcardPolicy: None
```

I am using AS3.

**Comment by mdditt2000** [comment]
Internal tracking PM Jira CONTCNTR-1732

**Comment by mdditt2000** [comment]
Permalink Edit Delete
amkgupta
Amit Kumar GUPTA added a comment - 16/Apr/20 2:01 AM
Github PR created to merge this fix.
CIS Sprint 37 ended 21/Apr/20

Closing issue. Fix in CIS 2.0
